lib/synapse/service_watcher/zookeeper.rb

```diff
@@ -41,6 +41,21 @@ def validate_discovery_opts
         unless @discovery['hosts']
       raise ArgumentError, "invalid zookeeper path for service #{@name}" \
         unless @discovery['path']
+      if @discovery['decode']
```

Comment:
I think we need to set @decode_method even if we don't have a decode (i.e. the default is zookeeper)